### ISSUE TITLE
[com.greengot] Impossible to use my banking app

### ISSUE THREAD

**Issue by qfl45** [microg/GmsCore#2017]
**Affected app**
Name: [Green-Got]
Package id: [com.greengot]

**Describe the bug**
Impossible to use my banking app. I will talk about my case in more details, but I've seen similar report and  I suppose that there is some other app which have the same problem. It is a problem that didn't exist several months ago, but it appeared maybe with app updates or MicroG updates or android updates, I don't know.

Just after logging into my account, just when the app is supposed to let me access the rest of the app, it gets stuck on the login screen and seems to keep trying to load indefinitely. Exactly the same thing happens when I open the application after closing it, except this time it's on the launch screen that the application gets stuck. So, impossible to access to the app and my banking account on my phone. As a result, to use this application, I have to use it on an Android tablet with Google Play Services. And it's by using it on this tablet that I've been able to observe certain differences and understand a little better exactly what the problem is, and I suppose that it has to do with authorizations and access to the Telephon and SMS from the application and/or MicroG. The best way to explain it is to show you with pictures, in the screenshot part.

**To Reproduce**
1. Just login or launch the app after login
2. Get stuck

**Expected behavior**
If the problem com from MicroG, I'm not an expert but I suppose it, it would be great if this problem could be patched and get my application working properly again and get access to the entire app and my banking account, and by the way the same thing for several others apps which have the same problem.

**Screenshots**
In these pics, it show what happen when the app is launched on the android tablet device with Google Play Services, this don't happen on my phone with MicroG on which the app just stay stuck, and show, I think, the problem on MicroG. In these pics, it happen at launching of the app after login and closing the app, but it is exactly the same thing just after login.

So, just after launching the app, it stay stuck on the launching screen (just like on my phone with MicroG) and I receive a notification, as you can see below.
![green-got_pic_1](https://github.com/microg/GmsCore/assets/143300163/2c2806e4-828b-4620-b286-56a09c6a623f)

After clicking on this notification, this small window appears over the app.
![green-got_pic_2](https://github.com/microg/GmsCore/assets/143300163/202d7420-bfbd-4ec3-bd68-7b912b1162bb)

As requested in this window, I give the necessary Telephone and SMS access authorizations to Google Play Services in the settings page of the Google Play Services application. And just after that, the app work very well without any problem. And as long as I don't grant these authorizations, the application remains blocked indefinitely, just like on my phone with MicroG. That seem to be logical if it need some authorizations.

BUT it work only with Google Play Services, even if I allow the Telephone and SMS access to MicroG or all the access available for MicroG on his setting page, the app stay stuck at the launching screen or just after login.

**System**
Android Version: [13]
Custom ROM: [LineageOS 20]

**microG**
microG Core version: [0.2.28.231657]
microG Self-Check results: [All ticked]

**Additional context**
Because of this problem, I'm forced to use an other android device, which is not my phone but a tablet, with Google Play Services, only to use my banking app and to be able to access to my banking account, it's very annoying to not be able to use my banking app on my phone. And I've seen than I'm not alone in this situation.

And also, before this bug happened only on launching screen and not at login.

**Comment by ale5000-git** [comment]
Hi,
there is already an implementation here: https://github.com/microg/GmsCore/commit/03836ce6dab2117b5d1c6d342bfa3a00fa517f37
but I'm not sure if it is already working completely for your app and there isn't a new version of microG yet.

**Comment by qfl45** [comment]
Hello,
So if I understand well, this problem is already being solved, and I just need to wait the next update of MicroG ?

**Comment by qfl45** [comment]
Okay, thank you.

**Comment by qfl45** [comment]
Hello,
I updated MicroG to v0.2.29.233013, but my banking app still don't working, exactly like with the previous version.

**Comment by breversa** [comment]
That's weird: I'm using MicroG `0.3.0.233515-129 (f11ed71)` integrated with [/e/OS](https://e.foundation/e-os/), and I don't have any issue with Green-Got... :-/

**Comment by qfl45** [comment]
I just seen you comment. Since you had comment about this, have you updated to latest microg ? Because I did it and Green-Got still doesn't work. If in your case it still work with updates, I'm wondering if there's a link with /e/OS. Because before I was on /e/OS, then I switched to LineageOS, and I remember I had never any problems on /e/OS with Green-Got, and I slowly began to have problems when I switched to LineageOS. So what you said seems to confirm it. If it is true, I wonder why is there this difference, mostly than /e/OS is based on LineageOS.
